src/test/java/org/wildfly/security/sasl/test/StringPrepTest.java

```diff
@@ -837,16 +837,16 @@ public void testForbidUnassigned() throws Exception {
     // ---------------------- helpers ----------------------
 
     private String codePointToString(int codePoint) {
-        ByteStringBuilder b = new ByteStringBuilder();
-        b.appendUtf8Raw(codePoint);
-        return new String(b.toArray(), StandardCharsets.UTF_8);
+        StringBuffer sb = new StringBuffer();
```

Comment:
We are now testing if StringBuffer is able to produce correct input for StringPrep rather than test our own ByteStringBuilder.appendUtf8Raw(int codePoint) method. @honza889 can you check what is wrong with appendUtf8Raw method in JDK8?